In [1]:
import obspy as op
import numpy as np
import cartopy.crs as ccrs
from obspy.clients.fdsn import Client
import cartopy.feature as cfeature

import matplotlib.pyplot as plt
%matplotlib inline
%config InlineBackend.figure_format = 'retina'

import datetime

In [2]:
currentDT = datetime.datetime.now()
print ('Download initiated. Should take about 8 seconds but requires an internet connection.')
print ('Download began: ',str(currentDT))

client = Client("IRIS")
t1 = op.UTCDateTime("1970-01-01T00:00:00") #start time of the request
t2 = op.UTCDateTime("2023-10-30T00:00:00") #end time of the request
minMag = 5
# read https://docs.obspy.org/packages/autogen/obspy.clients.fdsn.client.Client.get_events.html
catalog = client.get_events(starttime=t1, endtime=t2, minmagnitude=minMag)

currentDT = datetime.datetime.now()
print ('Download ended: ',str(currentDT))

Download initiated. Should take about 8 seconds but requires an internet connection.
Download began:  2023-10-30 19:16:14.117680
Download ended:  2023-10-30 19:20:46.429061


In [3]:
print(catalog)

92135 Event(s) in Catalog:
2023-10-29T13:57:41.061000Z | +43.941,  -28.384 | 5.1  mb
2023-10-29T12:28:17.941000Z | -30.670, -178.165 | 5.0  mb
...
1970-01-01T01:49:56.800000Z | +28.530, +129.390 | 5.1  mb
1970-01-01T01:43:48.000000Z |  +8.540,  -83.350 | 5.3  mb
To see all events call 'print(CatalogObject.__str__(print_all=True))'


In [4]:
# print the number of EQ and print info of the most recent one (has index=0)
print(len(catalog.events))
print(catalog[0].origins[0].latitude,catalog[0].origins[0].longitude)
print(catalog[0].origins[0].time)
print(catalog[0].magnitudes[0].mag)

92135
43.9406 -28.3835
2023-10-29T13:57:41.061000Z
5.1


In [5]:
# copy latitude, longitude, depth, and magnitude out of the catalog that they can be plotted below
evlat =[]
evlon =[]
evdep =[]
evmag = []
index = 0

for eve in catalog:
    event = catalog[index]
    org = event.origins
    evlat.append(org[0].latitude)
    evlon.append(org[0].longitude)
    #print(org[0].depth)
    if (org[0].depth==None):
        evdep.append(0.0)
    else:
        evdep.append((org[0].depth) / 1000.0)
    evmag.append((event.magnitudes[0].mag)**4/2)
    index+=1

In [6]:
#plt.rcParams['figure.figsize'] = [15, 7]
'''
fig = plt.figure(figsize=(50,30))

ax = fig.add_subplot(1, 1, 1, projection=ccrs.Mollweide(central_longitude = 0))

events = ax.scatter(evlon, evlat, c=evdep, s=evmag, transform=ccrs.PlateCarree())
    
#Add coastlines and gridlines for lat/lon
ax.coastlines(resolution='10m', color='gray')
ax.gridlines()

cbar = plt.colorbar(events, shrink=0.7)
cbar.set_label('Depth [km]', fontsize=50)
cbar.ax.tick_params(labelsize=40)

print('A world map with some earthquakes should appear. [Some warnings may also be printed.]')
plt.show()
'''

"\nfig = plt.figure(figsize=(50,30))\n\nax = fig.add_subplot(1, 1, 1, projection=ccrs.Mollweide(central_longitude = 0))\n\nevents = ax.scatter(evlon, evlat, c=evdep, s=evmag, transform=ccrs.PlateCarree())\n    \n#Add coastlines and gridlines for lat/lon\nax.coastlines(resolution='10m', color='gray')\nax.gridlines()\n\ncbar = plt.colorbar(events, shrink=0.7)\ncbar.set_label('Depth [km]', fontsize=50)\ncbar.ax.tick_params(labelsize=40)\n\nprint('A world map with some earthquakes should appear. [Some warnings may also be printed.]')\nplt.show()\n"

In [7]:
#question 2
#number of 8+ events since 1970 = 37 or 37/53 = 0.7 per year 

In [8]:
#question 3
#mag 9 = 2 events 
#mag 8 = 27 events 
#mag 7 = 571 events 
#mag 6 = 6243 events 
#mag 5 = 92135 events, run time increased to 5mins
#as mag decerases you can see the outline of plate boundaries and mid ocean ridges start to emerge 

In [9]:
#USA
USlongmin = -125.38316
USlongmax = -66.694649
USlatmin = 25.195198
USlatmax = 49.790668

#print(len(catalog))
USeq = 0

for i in range (len(catalog)):
    lat = catalog[i].origins[0].latitude
    long = catalog[i].origins[0].longitude 
    if USlatmin < lat and lat < USlatmax and USlongmin < long and long < USlongmax :
        USeq += 1
             
print('events in US:',USeq)
       


events in US: 712


In [10]:
#Japan
JPlongmin = 128.563941
JPlongmax = 146.600266
JPlatmin = 30.968076
JPlatmax = 45.634933

EqJP = 0

for i in range (len(catalog)):
    lat = catalog[i].origins[0].latitude
    long = catalog[i].origins[0].longitude 
    
    if JPlatmin < lat < JPlatmax and JPlongmin < long < JPlongmax :
        EqJP += 1
        
print('events in Japan:', EqJP)

events in Japan: 4474


| Magnitude | Total Number EQ | EQ in US | EQ in Japan | EQ elsewhere|
|---|---|---|---|---|
|Mag >= 9 | 9 | 0 | 1 | 8 |
|Mag >= 8 | 27 | 0 | 2 | 25 |
|Mag >= 7 | 571 | 6 | 31 | 534 |
|Mag >= 6 | 6243 | 388 | 64 | 5791 |
|Mag >= 5 | 92135 | 712 | 4474 | 86949 | 
